### DISCUSSION TITLE
Language not detected in repository (C++)

### DISCUSSION THREAD

**Discussion by b1scoito** [github-linguist/linguist#6241] [Classification] [answered]
Since I've created [this](https://github.com/b1scoito/external) repository, linguist doesn't seem to be detecting the language of my repository, that is C++.

I've tried setting the `linguist-detectable` flag on my `.gitattributes` file for all .cpp and .hpp files, but to no success. I have no idea what might be causing this, it just doesn't show any language on my repository.

Linguist appears to just not walk the directories that contain my C, C++ files, and it doesn't show any output. In the picture above I compare two projects with similar directory structure.
![image](https://user-images.githubusercontent.com/17802984/211101763-b3625f1c-e80c-4082-a3c5-d64ca3935403.png)

**Comment by lildude** [comment, marked as answer]
It's because all your code is in the `external` directory which Linguist considers as a path to vendored code here:

https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/vendor.yml#L111

You'll need to implement the `-linguist-vendored` override for these to be considered.

**Reply by b1scoito** [reply]
It worked! I had no idea that external was something that linguist considered as vendored code, thanks.
